### PR TITLE
Update assistant documentation guidance

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch updates the repository's assistant and documentation guidance from the latest upstream agent template and documentation style guide. It renders the root assistant instructions without the template-only Rust behaviour, adds canonical repository layout documentation, and introduces scoped guidance for the PWA, shared packages, and security automation directories.

No issue or roadmap task is linked to this documentation maintenance branch.

## Review walkthrough

- Start with [AGENTS.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/AGENTS.md) to review the rendered repository-wide assistant instructions.
- Review [docs/documentation-style-guide.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/docs/documentation-style-guide.md) for the imported upstream documentation conventions.
- Check [docs/contents.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/docs/contents.md) and [docs/repository-layout.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/docs/repository-layout.md) for documentation navigation and repository orientation.
- Finish with [frontend-pwa/AGENTS.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/frontend-pwa/AGENTS.md), [packages/AGENTS.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/packages/AGENTS.md), and [security/AGENTS.md](https://github.com/leynos/wildside/blob/chore/agent-docs-guidance/security/AGENTS.md) for the subdirectory-specific TypeScript and JavaScript guidance.

## Validation

- `make markdownlint`: passed.
- `make nixie`: passed.

## Notes

- `make nixie` refreshed `bun.lock` as a side effect of `bun install`; that unrelated lockfile churn was restored before committing.
- The pull request is opened as a draft by default.

### CODE REVIEW

File: frontend-pwa/AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- Provide localizations for all supported languages, including right-to-left
+  (RTL) support.
+- The site must use semantic classes.
+
+## Testing expectations
+
+- Ensure new features are validated with unit tests using `bun:test` where
+  applicable. Cover happy paths, unhappy paths, and relevant edge cases.
+- Ensure new features are validated with behavioural tests using `bun:test` and
+  `@aboviq/bun-test-cucumber` for Gherkin file support where applicable. Cover
```
[the remark this answers is not in the document]
Comment (leynos) [reply frontend-pwa/AGENTS.md:44]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

[the remark this answers is not in the document]
Comment (leynos) [reply frontend-pwa/AGENTS.md:44]: @coderabbitai Suggesting that developers don't know to add a dependency in order to use it is a bit cheeky

I think we can consider this closed.
